frontend/ui/uimanager.lua

```diff
@@ -47,8 +48,42 @@ function UIManager:init()
         -- resume.
         self:_initAutoSuspend()
         self.event_handlers["Suspend"] = function(input_event)
-            self:_stopAutoSuspend()
-            Device:onPowerEvent(input_event)
+            if not self._power_off_triggered then
```

Comment:
Pressing the power button (when not in `screen_saver_mode`) creates a `PowerPress` event which in UIManager sets `_power_off_triggered` to false and schedules `showPowerOffDialog` in 3 seconds. `showPowerOffDialog` sets `_power_off_triggered` to true and asks the user for confirmation. Releasing the power button creates a `Suspend` event, which `UIManager` doesn't do anything about if `_power_off_triggered` is true. This allows the power button to trigger 2 different actions:
* if it is released more than 3 seconds after being pressed, ask for confirmation of the power off and does not suspend
* if it is released less than 3 seconds after being pressed, suspend

Maybe I misunderstand something, but I think deleting `self.event_handlers["Suspend"]` would be a permanent change and break future suspend requests if the user cancels the power off request.